[AWS SSM] install the agent (#4273)
DataDog/documentation: content/en/agent/faq/install-the-agent-with-aws-ssm.md

---
title: Install the Agent with AWS SSM
kind: faq
further_reading:
- link: "agent/"
  tag: "Documentation"
  text: "Learn more about the Datadog Agent"
---

> Follow the steps below to install the Datadog Agent on an EC2 instance with AWS Systems Manager.

Reviewer: Might add a note about this being an "example configuration for a single manual instance being deployed". Beyond launching the instance, the rest of the steps `should` apply to autoscaling infrastructure, etc...

> Follow the steps below to install the Datadog Agent on an EC2 instance with AWS Systems Manager. **Note**: The EC2 step is for a single manual instance. The other steps can be used for multiple instances.

Reviewer 2: ```suggestion
Follow the steps below to install the Datadog Agent on an EC2 instance with AWS Systems Manager. 
**Note**: The EC2 step is for a single manual instance. Use the other steps for multiple instances.
```

### EC2 instance
* Create an [EC2 IAM Role][1] with the `AmazonEC2RoleforSSM` prebuilt policy or update your current [AWS Datadog policy][2].
* Launch an [EC2 instance][3] with this role attached.

### Parameter Store
In the [Parameter Store][4], create a parameter with:

> * Name: `dd-api-key-for-ssm`

Reviewer: Can be more generic/standard name, I chose that for identification while I was testing.

Author: Actually yours had your name in it. I changed it to this here and in the JSON. This way, the person following the instructions doesn't have to update the name in the JSON.

      "maxChars": 4096
    }
  },
  "mainSteps": [
    {
      "action": "aws:runShellScript",
      "name": "install_datadog",
      "precondition": {
        "StringEquals": [
          "platformType",
          "Linux"
        ]
      },
      "inputs": {
        "runCommand": [
          "wget -O ddinstall.sh https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh",
          "export DD_API_KEY=$(aws --region <AWS_REGION> ssm get-parameters --names dd-api-key-for-ssm --with-decryption --query Parameters[].Value --output text)",
          "bash ./ddinstall.sh"
        ],
        "workingDirectory": "{{ workingDirectory }}",
        "timeoutSeconds": "3600"
      }
    }
  ]
}
```

#### Run Command
Under [Run Command][7], click the **Run command** button and follow the steps below:

* **Command document**:
    * Click on the search box and select *Owner -> Owned by me*.
    * Click the radio button next to your document.
    * If necessary, choose the **Document version**.
* **Targets**:
    * Select the EC2 instance to target.
* **Output options** (optional):

>     * Select the **CloudWatch output** check box to log any issues.

Reviewer 2: ```suggestion
    * Select the **CloudWatch output** checkbox to log any issues.
```
